### PR TITLE
Feature: simple worker

### LINKED ISSUE #105: Non-Forking worker
**Is your feature request related to a problem? Please describe.**
Currently Qless-PHP ships with only a single worker implementation - `ForkingWorker`. In some environments the benefits presented by this model are outweighed by the issues with it. Notably, the Redis client library used throws a lot of errors when used with a forked process,  and processes using other stateful connections (e.g. PDO) also require special handling to ensure their connection is usable. In some cases these issues outweigh the benefits of a "managed" process - particularly if an external service manager is being used to ensure the worker is restarted when required.

**Describe the solution you'd like**
A "Simple" worker implementation that doesn't rely on forking to process jobs.

**Describe alternatives you've considered**
The project already attempts to 'hide' the errors from the Redis library, but numerous disconnects are still encountered in a project where I've used Qless-PHP. Re-connecting to a PDO data source in an `AfterFork` event handler is a partial workaround for PDO connections used by workers.
**Additional context**
This is something I've started working on in a fork.

I'd like to get your feedback/thoughts about adding this. As part of what I've written so far, I've also further abstracted some of the common logic from the existing `ForkingWorker` and the new `SimpleWorker` into traits - one that just handles the 'resource limits' handling (time, jobs, memory), and one that implements the basic 'job loop', with entry points for the individual implementations to e.g. fork, call the job performer directly, etc.

### PR DESCRIPTION
**Pull request by stephenreay**
Hello!

* Type: new feature
* Link to issue: #105 

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

TLDR: Adds a non-forking worker.

This separates a lot of the work-loop boilerplate logic currently in `ForkingWorker` into `JobLoopWorkerTrait`, pulls out the resource limiting logic into `ResourceLimitedWorkerTrait` and adds `ResourceLimitedWorkerInterface`.

Closes #105

Thanks

### CODE REVIEW

**Review by Slatch — DISMISSED** [review]

**Comment by sergeyklay** [comment]
I apologize @stephenreay for completely overlooking this. Thank you for the patch. This branch has conflicts that must be resolved (see https://github.com/pdffiller/qless-php/pull/131). Could you please rebase? Thanks again for the patch, and again I am sorry about the delay.

**Review by sergeyklay — APPROVED** [review]

**Comment by sergeyklay** [comment]
We will keep this open for a some short time to allow other contributors a chance to speak up, and to take a fresh look at it later. But at the moment it seems good to me. Thank you for the patch, and for helping us make Qless PHP better!

/cc @osadchyi-s
